fmutils.go

```diff
@@ -77,7 +77,20 @@ func (mask NestedMask) Filter(msg proto.Message) {
 				return true
 			}
 
-			if fd.IsList() {
+			if fd.IsMap() {
+				xmap := rft.Get(fd).Map()
+				xmap.Range(func(mk protoreflect.MapKey, mv protoreflect.Value) bool {
+					if mi, ok := m[mk.String()]; ok {
+						if i, ok := mv.Interface().(protoreflect.Message); ok && len(mi) > 0 {
+							mi.Filter(i.Interface())
+						}
```

Comment:
Please, update the type to `proto.Message` to stay consistent with the function signature above (same goes for `Prune()` function). 
```suggestion
						if i, ok := mv.Interface().(proto.Message); ok && len(mi) > 0 {
							mi.Filter(i)
						}
```

Reply:
I've had it like this, but your version does not actually work because the type cast to `proto.Message` fails. Did you try this, do I miss anything? :thinking:

Reply:
Hm.. I've just tried it and it worked. Could you share the error message that you are getting? (btw, my go version is 1.18)

Reply:
I tried it with go 1.15, 1.16, 1.17 and 1.18 and it worked

Reply:
The unit tests still fail for me. I pushed the change now regardless, just to check what CI thinks about it, and as you can see, it also fails. I'm confused now.

Reply:
Hm.. I did not run your tests, u just compiled it and it did not complain. I'll take a look at the tests now

Reply:
Ah :joy:  Yes, it does compile, that's not the problem.

Reply:
FTR, the version I pushed originally today makes the tests go green.

Reply:
Ok, let's use `protoreflect.Message` as `mv.Interface()` is apparently `protoreflect.Message` and not `proto.Message`.

Reply:
I don't really get it though because

https://github.com/protocolbuffers/protobuf-go/blob/master/proto/proto.go:

```
package proto
...
type Message = protoreflect.ProtoMessage
```

So the interface should be congruent and hence the type cast should succeed.

Reply:
I don't get it either. From my perspective those 2 interfaces should be interchangeable, but it looks they are not.